website/addons/evernote/tests/test_views.py

```diff
@@ -0,0 +1,236 @@
+    def test_oauth_start(self):
+        pass
+
+    def setUp(self):
+        self.mock_refresh = mock.patch("website.addons.evernote.model.Evernote.refresh_oauth_key")
+        self.mock_refresh.return_value = True
+        self.mock_refresh.start()
+        super(TestAuthViews, self).setUp()
+
+    def tearDown(self):
+        self.mock_refresh.stop()
+        super(TestAuthViews, self).tearDown()
+
+    @mock.patch(
+        'website.addons.box.model.BoxUserSettings.revoke_remote_oauth_access',
+        mock.PropertyMock()
+    )
+    def test_delete_external_account(self):
+        super(TestAuthViews, self).test_delete_external_account()
+
+class TestConfigViews(EvernoteAddonTestCase, testing.views.OAuthAddonConfigViewsTestCaseMixin):
+
+    folder = {
+        'path': '/Foo',
+        'id': '12234'
+    }
+    Serializer = EvernoteSerializer
+    client = mock_client
+
+    def setUp(self):
+        self.mock_data = mock.patch.object(
+            EvernoteNodeSettings,
+            '_folder_data',
+            return_value=(self.folder['id'], self.folder['path'])
+        )
+        self.mock_data.start()
+        super(TestConfigViews, self).setUp()
+
+    def tearDown(self):
+        self.mock_data.stop()
+        super(TestConfigViews, self).tearDown()
+
+    @mock.patch.object(EvernoteSerializer, 'credentials_are_valid', return_value=True)
+    def test_import_auth(self, *args):
+        super(TestConfigViews, self).test_import_auth()
+
+class TestFilebrowserViews(EvernoteAddonTestCase):
+
+    def setUp(self):
+        super(TestFilebrowserViews, self).setUp()
+        self.user.add_addon('evernote')
+        self.node_settings.external_account = self.user_settings.external_accounts[0]
+        self.node_settings.save()
+        self.patcher_refresh = mock.patch('website.addons.evernote.model.Evernote.refresh_oauth_key')
+        self.patcher_refresh.return_value = True
+        self.patcher_refresh.start()
+
+    def tearDown(self):
+        self.patcher_refresh.stop()
+
+    def test_evernote_list_folders(self):
+        with patch_client('website.addons.evernote.model.Evernote'):
+            url = self.project.api_url_for('evernote_folder_list', folder_id='foo')
+            print('TestFilebrowserViews.test_evernote_list_folders-->url: ', url)
+
+            print('TestFilebrowserViews.test_evernote_list_folders-->app: ', self.app)
+
+            res = self.app.get(url, auth=self.user.auth)
+            print('TestFilebrowserViews.test_evernote_list_folders-->res.json: ', res.json)
+
+            contents = mock_client.get_folder('', list=True)
+            print('TestFilebrowserViews.test_evernote_list_folders-->contents: ', contents)
+
+            # comparing EvernoteAddonTestCase with mock_client (MockEvernote)
+            assert_equal( res.json[0]['name'], contents['name'])
+
+
+            # THE REST IS EXTRANEOUS, I THINK
```

Comment:
All the rest of this file is commented out code adapted from tests in the box add-on.  I've not deleted this section because I might need to expand the tests I currently have.